app/org/sagebionetworks/bridge/dynamodb/DynamoSubpopulationDao.java

```diff
@@ -0,0 +1,147 @@
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBQueryExpression;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper.FailedBatch;
+import com.google.common.collect.Lists;
+
+@Component
+public class DynamoSubpopulationDao implements SubpopulationDao {
+    
+    private DynamoDBMapper mapper;
+
+    /** DynamoDB mapper for the HealthDataRecord table. This is configured by Spring. */
+    @Resource(name = "subpopulationDdbMapper")
+    public void setMapper(DynamoDBMapper mapper) {
+        this.mapper = mapper;
+    }
+    
+    @Override
+    public Subpopulation createSubpopulation(Subpopulation subpop) {
+        checkNotNull(subpop);
+        checkNotNull(subpop.getStudyIdentifier());
+
+        subpop.setVersion(null);
+        subpop.setDeleted(false);
+        subpop.setGuid(BridgeUtils.generateGuid());
+        mapper.save(subpop);
+        return subpop;
+    }
+
+    @Override
+    public Subpopulation updateSubpopulation(Subpopulation subpop) {
+        checkNotNull(subpop);
+        checkNotNull(subpop.getStudyIdentifier());
+        checkNotNull(subpop.getVersion());
+        checkNotNull(subpop.getGuid());
+        
+        // This will throw a not found exception if this is not an update. 
+        // It also prevents deletion through this call.
+        Subpopulation existing = getSubpopulation(new StudyIdentifierImpl(subpop.getStudyIdentifier()), subpop.getGuid());
+        subpop.setDeleted(existing.isDeleted());
+        
+        mapper.save(subpop);
+        return subpop;
+    }
+
+    @Override
+    public List<Subpopulation> getSubpopulations(StudyIdentifier studyId, boolean allExistingRecordsOnly) {
+        DynamoSubpopulation hashKey = new DynamoSubpopulation();
+        hashKey.setStudyIdentifier(studyId.getIdentifier());
+        
+        DynamoDBQueryExpression<DynamoSubpopulation> query = 
+                new DynamoDBQueryExpression<DynamoSubpopulation>().withHashKeyValues(hashKey);
+        
+        // Get all the records because we only create a default if there are no physical records, 
+        // regardless of the deletion status.
+        List<DynamoSubpopulation> subpops = mapper.query(DynamoSubpopulation.class, query);
+        if (!allExistingRecordsOnly && subpops.isEmpty()) {
+            Subpopulation subpop = createDefaultSubpopulation(studyId);
+            return Lists.newArrayList(subpop);
+        }
+        // Now filter out deleted subpopulations, if requested
+        return subpops.stream()
+            .filter(subpop -> allExistingRecordsOnly || !subpop.isDeleted())
+            .sorted(SPECIFICITY_SORTER).collect(Collectors.toList());
+    }
+    
+    @Override
+    public Subpopulation createDefaultSubpopulation(StudyIdentifier studyId) {
+        DynamoSubpopulation subpop = new DynamoSubpopulation();
+        subpop.setStudyIdentifier(studyId.getIdentifier());
+        subpop.setGuid(studyId.getIdentifier());
```

Comment:
No, for backwards compatibility with the existing consent documents, we're setting this to study ID. This means the default subpopulation will retrieve the existing consents, and existing studies won't lose this. Later subpopulations will use a different string, a GUID.